server/metric_reporter.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/plugin/segment"
+
+	"go.uber.org/zap"
+)
+
+const (
+	metricSchedulerInterval = time.Duration(24) * time.Hour
+)
+
+// MetricScheduler is the metric scheduler.
+type MetricScheduler struct {
+	l *zap.Logger
+
+	server  *Server
+	metrics api.MetricService
+}
+
+// NewMetricScheduler creates a new metric scheduler.
+func NewMetricScheduler(logger *zap.Logger, server *Server, deploymentID string) *MetricScheduler {
+	segmentService := segment.NewService(logger, server.profile.SegmentKey, deploymentID, server.store)
+
+	plan := api.FREE.String()
+	if server.subscription != nil {
+		plan = server.subscription.Plan.String()
+	}
+
+	segmentService.Identify(&api.Workspace{
+		Plan:         plan,
+		DeploymentID: deploymentID,
+	})
+
+	return &MetricScheduler{
+		l:       logger,
+		server:  server,
+		metrics: segmentService,
+	}
+}
+
+// Run will run the metric scheduler.
+func (m *MetricScheduler) Run(ctx context.Context, wg *sync.WaitGroup) {
+	ticker := time.NewTicker(metricSchedulerInterval)
+	defer ticker.Stop()
+	defer wg.Done()
+	m.l.Debug(fmt.Sprintf("Metrics scheduler started and will run every %v", metricSchedulerInterval))
+
```

Comment:
I suggest we add a delay for the first run. Otherwise, there will be quite a lot of database activities upon starting up Bytebase.

Reply:
I tested it in local, seems it won't trigger immediately as the interval is 24 hours?

Reply:
Yes, then we have another problem during restart (upgrade), we are likely to miss the 1 day data.

usually, we would collect the data upon startup and then wait for the next cycle.

Reply:
oh, nm, we change it to 1 hour, then the current code is fine